### PR TITLE
webui add remove user to groups not case sensitive

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

Add webui acceptance tests to:

- create users using different case of the username (UID)

- add and delete groups using different case of the group name, checking that distinct groups are being managed

- add and delete users in groups, using different case of group name, making sure that distinct groups are managed correctly, and the user names behave in the "opposite" way (they are not case-sensitive)

## Related Issue

part of https://github.com/owncloud/core/issues/35354

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by phil-davis** [comment]
@HariBhandari07 
- please click the buttons to "sign the CLA"
- rebase this PR to current user_management master branch (somehow it is already out-of-date)

**Comment by phil-davis** [comment]
@HariBhandari07 GitHub says that PR is out-of-date with master. You will need to locally pull and up-to-date user_management master branch, then rebase to that and force push.

File: tests/acceptance/features/webUIManageUsersGroups/addUsers.feature [hunk 1]
```diff
@@ -202,3 +202,23 @@ Feature: add users
     And the user follows the password set link received by "bnu@owncloud" using the webUI
     And the user sets the password to " " and confirms with the same password using the webUI
     Then the user should be redirected to a webUI page with the title "%productname%"
+
+  Scenario Outline: admin creates a user but logs in with different case username
+    Given user "brand-new-user" has been deleted
```
Comment (individual-it) [inline tests/acceptance/features/webUIManageUsersGroups/addUsers.feature]: no need to do that, in some tests its done but its legacy

File: tests/acceptance/features/webUIManageUsersGroups/addUsers.feature [hunk 2]
```diff
@@ -202,3 +202,23 @@ Feature: add users
...
+    And the user should be redirected to a webUI page with the title "Files - %productname%"
+    Examples:
+      | creation-username | login-username | username       |
+      | Brand-New-User    | brand-new-user | BRAND-NEW-USER |
+      | brand-new-user    | Brand-New-User | Brand-New-User |
+      | Brand-New-User    | BRAND-NEW-USER | brand-new-user |
+      | brand-new-user    | Brand-New-User | BRAND-NEW-USER |
+
+  Scenario: admin tries to create an existing user but with username containing capital letters
+    Given user "user1" has been created with default attributes
```
Comment (individual-it) [inline tests/acceptance/features/webUIManageUsersGroups/addUsers.feature]: create the user without skeleton to speed it up

File: tests/acceptance/features/webUIManageUsersGroups/editUsers.feature [hunk 3]
```diff
@@ -91,3 +91,22 @@ Feature: edit users
     When the administrator changes the email of user "user0" to "new_email@oc.com" using the webUI
     Then user "user0" should exist
     And the email address of user "user0" should be "new_email@oc.com"
+
+  Scenario Outline: remove a user from a group using mixes of upper and lower case in group names
+    Given user "user0" has been created with default attributes
```
Comment (individual-it) [inline tests/acceptance/features/webUIManageUsersGroups/editUsers.feature]: create the user without skeleton folder, to make the tests faster

File: tests/acceptance/features/bootstrap/WebUIUsersContext.php [hunk 4]
```diff
@@ -50,6 +50,12 @@ class WebUIUsersContext extends RawMinkContext implements Context {
 	 */
 	private $webUIGeneralContext;
 
+	/**
+	 *
+	 * @var GroupsCreated
+	 */
+	private $groupsCreated = [];
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIUsersContext.php]: use the created groups from featureContext see https://github.com/owncloud/core/blob/5fff7599545af9e365eda39af3c52798f0517208/tests/acceptance/features/bootstrap/Provisioning.php#L42
there is no `add` function so you will have to create one there

File: tests/acceptance/features/bootstrap/WebUIUsersContext.php [hunk 5]
```diff
@@ -192,6 +198,17 @@ public function theAdminDeletesDoesNotDeleteGroupUsingWebUI($name) {
...
 
+	/**
+	 * @When the administrator deletes group :groupName using the webUI
+	 *
+	 * @param string $groupName
+	 *
+	 * @return void
+	 */
+	public function theAdministratorDeletesGroupUsingTheWebui($groupName) {
+		$this->theAdminDeletesTheGroupUsingTheWebUI($groupName);
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIUsersContext.php]: why not using the existing step directly?

File: tests/acceptance/features/bootstrap/WebUIUsersContext.php [hunk 6]
```diff
@@ -563,8 +580,21 @@ public function restoreScenario(AfterScenarioScope $afterScenarioScope) {
 	 */
 	public function theAdminAddsGroupUsingTheWebUI($groupName) {
 		$this->usersPage->addGroup($groupName, $this->getSession());
+		\array_push($this->groupsCreated, $groupName);
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIUsersContext.php]: see comment to  `$this->groupsCreated`

File: tests/acceptance/features/bootstrap/WebUIUsersContext.php [hunk 7]
```diff
@@ -563,8 +580,21 @@ public function restoreScenario(AfterScenarioScope $afterScenarioScope) {
...
 	}
 
+	/**
+	 * @AfterScenario
+	 *
+	 * @return void
+	 * @throws \Exception
+	 */
+	public function cleanupGroups() {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIUsersContext.php]: let the featureContext deal with the cleanup https://github.com/owncloud/core/blob/5fff7599545af9e365eda39af3c52798f0517208/tests/acceptance/features/bootstrap/Provisioning.php#L2860
and even if you have to do it here, do not use the UI to clean-up, the API is much faster

File: tests/acceptance/features/webUIManageUsersGroups/addGroup.feature [hunk 8]
```diff
@@ -11,6 +11,36 @@ Feature: Add group
...
       | groupname   |
       | "localuser" |
+
+  Scenario Outline: group names are case-sensitive, multiple groups can exist with different upper and lower case names
+    When the administrator adds group "<group_id1>" using the webUI
+    And the administrator adds group "<group_id2>" using the webUI
+    Then the group name "<group_id1>" should be listed on the webUI
+    And the group name "<group_id2>" should be listed on the webUI
+    But the group name "<group_id3>" should not be listed on the webUI
+    And group "<group_id3>" should not exist
```
Comment (individual-it) [inline tests/acceptance/features/webUIManageUsersGroups/addGroup.feature]: please also check that the groups should exist

**Comment by individual-it** [comment]
backport to stable10 in https://github.com/owncloud/core/pull/35463

**Pull request merged**